app/controllers/api/v1/user_projects_controller.rb

```diff
@@ -23,11 +24,44 @@ def show
                include: :user
       end
 
+      def create
+        raise PermissionError.new(Project, :manage) unless can_manage_project?(@project)
+
+        user_project = @project.user_projects.create!(user_project_params.merge!(assigned_by: current_user))
+
+        render jsonapi: user_project, serializer: UserProjectSerializer, status: :created
+      end
+
+      def update
+        @user_project.role = user_project_params[:role]
+        return render body: nil, status: :no_content unless @user_project.changed?
+
+        @user_project.assigned_by = current_user
+        @user_project.save!
+        render jsonapi: @user_project, serializer: UserProjectSerializer, status: :ok
+      end
+
+      def destroy
+        @user_project.destroy!
```

Comment:
Should we check here if the user is not unassigning himself?

Reply:
Or at least check if he is owner (or the last user on a project).

Reply:
Do we need such checks in an API?

Reply:
I think it is better to move such check to the model itself

Reply:
true